corehq/apps/sms/mixin.py

```diff
@@ -138,26 +142,43 @@ def by_suffix(cls, phone_number, include_pending=False):
         Used to lookup a VerifiedNumber, trying to exclude country code digits.
         """
         try:
-            result = cls.phone_lookup(
-                "phone_numbers/verified_number_by_suffix",
-                phone_number,
-                include_pending
-            )
+            result = cls._by_suffix(apply_leniency(phone_number))
+            return cls._filter_pending(result, include_pending)
         except MultipleResultsFound:
             # We can't pinpoint who the number belongs to because more than one
             # suffix matches. So treat it as if the result was not found.
-            result = None
-        return result
+            return None
+
+    @classmethod
+    @quickcache(['phone_number'], timeout=60 * 60)
+    def _by_phone(cls, phone_number):
+        return cls.phone_lookup('phone_numbers/verified_number_by_number', phone_number)
+
+    @classmethod
+    @quickcache(['phone_number'], timeout=60 * 60)
+    def _by_suffix(cls, phone_number):
+        return cls.phone_lookup('phone_numbers/verified_number_by_suffix', phone_number)
 
     @classmethod
-    def phone_lookup(cls, view_name, phone_number, include_pending=False):
+    def phone_lookup(cls, view_name, phone_number):
         # We use .one() here because the framework prevents duplicates
         # from being entered when a contact saves a number.
         # See CommCareMobileContactMixin.save_verified_number()
-        v = cls.view(view_name,
-                     key=apply_leniency(phone_number),
-                     include_docs=True).one()
-        return v if (include_pending or (v and v.verified)) else None
+        return cls.view(
+            view_name,
+            key=phone_number,
+            include_docs=True
+        ).one()
+
+    @classmethod
+    def _filter_pending(cls, v, include_pending):
+        if v:
```

Comment:
yeah that looks like it would work, though i think the current state is a little easier to follow